spot_rl_experiments/spot_rl/envs/place_env.py

```diff
@@ -2,69 +2,249 @@
-    parser.add_argument("-w", "--waypoint")
-    parser.add_argument("-l", "--target_is_local", action="store_true")
-    args = parser.parse_args()
-    config = construct_config(opts=args.opts)
+    parser.add_argument(
+        "-p",
+        "--place_target",
+        help="input:float,float,float -> place target x,y,z in meters",
+    )
+    parser.add_argument(
+        "-w",
+        "--waypoints",
+        type=str,
+        help="input:string -> place target waypoints (comma separated place_target names) where robot needs to place the object",
+    )
+    parser.add_argument(
+        "-l",
+        "--target_is_local",
+        action="store_true",
+        help="whether the place target specified is in the local frame of the robot",
+    )
+    args = parser.parse_args(args=args)
+
+    return args
+
+
+def construct_config_for_place(file_path=None, opts=[]):
+    config = None
+    if file_path is None:
+        config = construct_config(opts=opts)
+    else:
+        config = construct_config(file_path=file_path, opts=opts)
 
     # Don't need cameras for Place
     config.USE_HEAD_CAMERA = False
     config.USE_MRCNN = False
 
-    if args.waypoint is not None:
-        assert not args.target_is_local
-        place_target = place_target_from_waypoints(args.waypoint)
-    else:
-        assert args.place_target is not None
-        place_target = [float(i) for i in args.place_target.split(",")]
-    env = SpotPlaceEnv(config, spot, place_target, args.target_is_local)
-    env.power_robot()
-    policy = PlacePolicy(config.WEIGHTS.PLACE, device=config.DEVICE)
-    policy.reset()
-    observations = env.reset()
-    done = False
-    env.say("Starting episode")
-    while not done:
-        action = policy.act(observations)
-        observations, _, done, _ = env.step(arm_action=action)
-    if done:
-        while True:
-            env.reset()
-            spot.set_base_velocity(0, 0, 0, 1.0)
+    return config
+
+
+class PlaceController:
+    """
+    Place controller is used to execute place for given place targets
+
+    Args:
+        config: Config object
+        spot: Spot object
+        use_policies (bool): Whether to use policies or use BD API to execute place
+
+    How to use:
+        1. Create PlaceController object
+        2. Call execute() with place_target_list as input
+        3. Call shutdown() to stop the robot
+
+    Example:
+        config = construct_config_for_place(opts=[])
+        spot = Spot("PlaceController")
+        place_target_list = [target1, target2, ...]
+        place_controller = PlaceController(config, spot, use_policies=True)
+        place_result = place_controller.execute(place_target_list)
+        place_controller.shutdown()
+    """
+
+    def __init__(self, config, spot: Spot, use_policies=True):
+        self.spot = spot
+        self.use_policies = use_policies
+
+        # Setup
+        self.policy = PlacePolicy(config.WEIGHTS.PLACE, device=config.DEVICE)
```

Comment:
Thanks! So, we still need weight when we do not use policy?

Reply:
Currently we are setting up the policy and env objects regardless.

But you are accurate. In future, we can also skip initializing the policy object. We can add that as an improvement.

Reply:
We updated this to only load policies if `use_policies` is True